core/src/test/scala/org/apache/spark/scheduler/DAGSchedulerSuite.scala

```diff
@@ -228,6 +232,9 @@ class DAGSchedulerSuite extends SparkFunSuite with LocalSparkContext with Timeou
   override def afterEach(): Unit = {
     try {
       scheduler.stop()
+      dagEventProcessLoopTester.stop()
+      mapOutputTracker.stop()
+      broadcastManager.stop()
```

Comment:
`broadcastManager` actually isn't leaking any threads given the current implemenation of `TorrentBroadcastFactory`, but still we should probably stop it.  Also worth noting that netty does *not* clean up the threads for the shuffle server and client immediately, so those still linger on between tests for a little while, but they do get cleaned up eventually.